core/templates/dev/head/pages/exploration_editor/feedback_tab/ThreadDataService.js

```diff
@@ -63,32 +63,22 @@ oppia.factory('ThreadDataService', [
           target_id: _expId
         };
       }
-      var sPromise = $http.get(_SUGGESTION_LIST_HANDLER_URL, {
+      var suggestionsPromise = $http.get(_SUGGESTION_LIST_HANDLER_URL, {
         params: params
       });
 
-      $q.all([fPromise, sPromise]).then(function(res) {
+      $q.all([threadsPromise, suggestionsPromise]).then(function(res) {
         _data.feedbackThreads = res[0].data.threads;
-        _data.suggestionThreads = [];
         if (constants.USE_NEW_SUGGESTION_FRAMEWORK) {
+          _data.suggestionThreads = res[0].data.threads_with_suggestions;
           for (var i = 0; i < res[1].data.suggestions.length; i++) {
             suggestion = res[1].data.suggestions[i];
             suggestion.thread_id = suggestion.suggestion_id.slice(
               suggestion.suggestion_id.indexOf('.') + 1);
-            _data.suggestionThreads.push(suggestion);
-          }
-          for (var i = 0; i < _data.suggestionThreads.length; i++) {
-            for (var j = 0; j < _data.feedbackThreads.length; j++) {
-              if (_data.suggestionThreads[i].thread_id ===
-                  _data.feedbackThreads[j].thread_id) {
-                _data.suggestionThreads[i].subject = (
-                  _data.feedbackThreads[j].subject);
-                _data.suggestionThreads[i].description = (
-                  _data.feedbackThreads[j].description);
-                _data.feedbackThreads.splice(j, 1);
-                // As only one feedback thread can link to one suggestion thread
-                // We can stop the inner loop when we find a match and move on
-                // to the next suggestion thread.
+            for (var j = 0; j < _data.suggestionThreads.length; j++) {
+              if (suggestion.thread_id ===
+                  _data.suggestionThreads[j].thread_id) {
+                _data.suggestionThreads[j].suggestion = suggestion;
```

Comment:
Oh, no -- you should not be attaching properties to a dict midstream. This won't work.

If you're going to do this we need a SuggestionThread domain object and it needs to contain a suggestion subfield whose value is a Suggestion object. Alternatively, keep Suggestion and SuggestionThread objects separate and find a way to get the suggestion corresponding to a thread "on the fly".

Reply:
Fixed by using domain objects.